app/controllers/flowcommerce_spree/inventory_controller.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module FlowcommerceSpree
+  class InventoryController < ActionController::Base
+    def stock
+      response = []
+      params['items'].each { |data| response << check_stock(data[:id], data[:qty].to_i) }
+
+      render json: { items: response }, status: :ok
+    end
+
+    private
+
+    def check_stock(flow_id, quantity)
+      variant = Spree::Variant.find_by("meta -> 'flow_data' ->> 'id' = ?", flow_id)
```

Comment:
@sebastiandl , I saw the param in task is described as flow_number, not flow_id. If so, then it is just the variant.sku - much simpler.

Reply:
Wow thanks for the heads up!!! I should have asked (wrong assumption from my side) I ll fix this!